test/notebook_tests/test_notebooks.py

```diff
@@ -0,0 +1,86 @@
+import fileinput
+import os
+import logging
+from shutil import copyfile
+
+import nbformat
+import pytest
+from nbconvert.preprocessors import ExecutePreprocessor
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+demo_path = "examples/"
+demo_notebooks = []
+
+
+for dir_, _, files in os.walk(demo_path):
+    for file_name in files:
+        rel_file = os.path.join(dir_, file_name)
+        if file_name.endswith(".ipynb") and "QPU" not in file_name:
+            demo_notebooks.append((dir_, rel_file))
+
+
+def rename_bucket(notebook_path, s3_bucket):
+    with fileinput.FileInput(notebook_path, inplace=True) as file:
+        for line in file:
+            print(
+                line.replace("amazon-braket-Your-Bucket-Name", s3_bucket).replace(
+                    "amazon-braket-<bucket name>", s3_bucket
+                ),
+                end="",
+            )
+
+
+def check_tn1_availability(notebook_path, region):
+    tn1_device = "arn:aws:braket:::device/quantum-simulator/amazon/tn1"
+    tn1_regions = ["us-west-2", "us-east-1"]
```

Comment:
I'd genericize this to `_check_exclusive_device_availability`, and then extract a dict:
```python
_EXCLUSIVE_DEVICE_REGIONS = {
    "arn:aws:braket:::device/quantum-simulator/amazon/tn1": ["us-west-2", "us-east-1"]
}
```
This way, we'll be future proof if we ever get other region-exlusive devices.